iOS error compilation version facebook_app_events: ^0.13.0

Command CompileSwift failed with a nonzero exit code
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:119:81: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["email"] as? String, forType: AppEvents.UserDataType.email)
                                                                          ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:120:85: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["firstName"] as? String, forType: AppEvents.UserDataType.firstName)
                                                                              ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:121:84: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["lastName"] as? String, forType: AppEvents.UserDataType.lastName)
                                                                             ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:122:81: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["phone"] as? String, forType: AppEvents.UserDataType.phone)
                                                                          ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:123:87: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["dateOfBirth"] as? String, forType: AppEvents.UserDataType.dateOfBirth)
                                                                                ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:124:82: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["gender"] as? String, forType: AppEvents.UserDataType.gender)
                                                                           ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:125:80: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["city"] as? String, forType: AppEvents.UserDataType.city)
                                                                         ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:126:81: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["state"] as? String, forType: AppEvents.UserDataType.state)
                                                                          ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:127:79: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["zip"] as? String, forType: AppEvents.UserDataType.zip)
                                                                        ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:128:83: error: type 'AppEvents' has no member 'UserDataType'
            AppEvents.setUserData(arguments["country"] as? String, forType: AppEvents.UserDataType.country)
                                                                            ~~~~~~~~~ ^~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:143:19: error: type 'AppEvents' has no member 'updateUserProperties'
            AppEvents.updateUserProperties( parameters, handler: { (connection, response, error) in
            ~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~
    /Users/assert/.pub-cache/hosted/pub.dartlang.org/facebook_app_events-0.13.0/ios/Classes/SwiftFacebookAppEventsPlugin.swift:187:43: error: 'nil' is not compatible with expected argument type 'ApplicationDelegate'
            ApplicationDelegate.initializeSDK(nil)
                                              ^
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

--- a/ios/Classes/SwiftFacebookAppEventsPlugin.swift
+++ b/ios/Classes/SwiftFacebookAppEventsPlugin.swift
@@ -116,16 +116,16 @@ public class SwiftFacebookAppEventsPlugin: NSObject, FlutterPlugin {
 
     private func handleSetUserData(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         let arguments = call.arguments as? [String: Any] ?? [String: Any]()
-        AppEvents.setUserData(arguments["email"] as? String, forType: AppEvents.UserDataType.email)
-        AppEvents.setUserData(arguments["firstName"] as? String, forType: AppEvents.UserDataType.firstName)
-        AppEvents.setUserData(arguments["lastName"] as? String, forType: AppEvents.UserDataType.lastName)
-        AppEvents.setUserData(arguments["phone"] as? String, forType: AppEvents.UserDataType.phone)
-        AppEvents.setUserData(arguments["dateOfBirth"] as? String, forType: AppEvents.UserDataType.dateOfBirth)
-        AppEvents.setUserData(arguments["gender"] as? String, forType: AppEvents.UserDataType.gender)
-        AppEvents.setUserData(arguments["city"] as? String, forType: AppEvents.UserDataType.city)
-        AppEvents.setUserData(arguments["state"] as? String, forType: AppEvents.UserDataType.state)
-        AppEvents.setUserData(arguments["zip"] as? String, forType: AppEvents.UserDataType.zip)
-        AppEvents.setUserData(arguments["country"] as? String, forType: AppEvents.UserDataType.country)
+        AppEvents.setUserData(arguments["email"] as? String, forType: FBSDKAppEventUserDataType.email)
+        AppEvents.setUserData(arguments["firstName"] as? String, forType: FBSDKAppEventUserDataType.firstName)
+        AppEvents.setUserData(arguments["lastName"] as? String, forType: FBSDKAppEventUserDataType.lastName)
+        AppEvents.setUserData(arguments["phone"] as? String, forType: FBSDKAppEventUserDataType.phone)
+        AppEvents.setUserData(arguments["dateOfBirth"] as? String, forType: FBSDKAppEventUserDataType.dateOfBirth)
+        AppEvents.setUserData(arguments["gender"] as? String, forType: FBSDKAppEventUserDataType.gender)
+        AppEvents.setUserData(arguments["city"] as? String, forType: FBSDKAppEventUserDataType.city)
+        AppEvents.setUserData(arguments["state"] as? String, forType: FBSDKAppEventUserDataType.state)
+        AppEvents.setUserData(arguments["zip"] as? String, forType: FBSDKAppEventUserDataType.zip)
+        AppEvents.setUserData(arguments["country"] as? String, forType: FBSDKAppEventUserDataType.country)
 
         result(nil)
     }

--- a/ios/Classes/SwiftFacebookAppEventsPlugin.swift
+++ b/ios/Classes/SwiftFacebookAppEventsPlugin.swift
@@ -10,7 +10,7 @@ public class SwiftFacebookAppEventsPlugin: NSObject, FlutterPlugin {
         // Required for FB SDK 9.0, as it does not initialize the SDK automatically any more.
         // See: https://developers.facebook.com/blog/post/2021/01/19/introducing-facebook-platform-sdk-version-9/
         // "Removal of Auto Initialization of SDK" section
-        instance.initializeSDK()
+        ApplicationDelegate.initializeSDK()
 
         registrar.addMethodCallDelegate(instance, channel: channel)
     }
@@ -181,9 +181,5 @@ public class SwiftFacebookAppEventsPlugin: NSObject, FlutterPlugin {
         let enabled = arguments["enabled"] as! Bool
         Settings.setAdvertiserTrackingEnabled(enabled)        
         result(nil)
-    }
-
-    public func initializeSDK() {
-        ApplicationDelegate.initializeSDK(nil)
     }
 }

--- a/ios/facebook_app_events.podspec
+++ b/ios/facebook_app_events.podspec
@@ -13,7 +13,7 @@ Flutter plugin for Facebook Analytics and App Events
   s.public_header_files = 'Classes/**/*.h'
   s.dependency 'Flutter'
   s.dependency 'FBSDKCoreKit', '~> 11.1.0'
-  s.swift_version       = '4.0'
+  s.swift_version       = '5.0'
 
   s.ios.deployment_target = '9.0'
 end

